### PR TITLE
Run new version of BioCro on Setaria params

### PR DESCRIPTION
**Pull request by KristinaRiemer**

### CODE REVIEW

File: BioCro/DARPA/biocro_1.0_darpa.Rmd [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+```{r, eval=FALSE}
+install.packages('biocro_0.95/biocro-master', repos = NULL, type = 'SOURCE')
+```
+
+BioCro 1.0: 
+```{r, message=FALSE}
+install.packages('biocro_1.00/biocro-new-framework', repos = NULL, type = 'SOURCE')
+```
+
+Check version:
```
Comment (Chris-Schnaufer) [inline BioCro/DARPA/biocro_1.0_darpa.Rmd:22]: Perhaps have this first so that it can be checked then downloaded?

File: BioCro/DARPA/biocro_1.0_darpa.Rmd [hunk 2]
```diff
@@ -0,0 +1,246 @@
...
+OpBioGro_weather <- read.csv("biocro_opt_darpa_files/OpBioGro_weather.csv") %>% 
+  rename(solar = solarR, temp = DailyTemp.C, rh = RH, windspeed = WindSpeed)
+ OpBioGro_biomass <- read.csv("biocro_opt_darpa_files/OpBioGro_biomass.csv")
+```
+
+### Set up parameters
+
+The following code uses BioCro 1.0. 
+
+Set up the parameters for *Setaria* with these two lists, `setaria_initial_state` and `setaria_parameters`.
```
Comment (Chris-Schnaufer) [inline BioCro/DARPA/biocro_1.0_darpa.Rmd:41]: Maybe reword: "... with these two ..." to "... using these two ..."

File: BioCro/DARPA/biocro_1.0_darpa.Rmd [hunk 3]
```diff
@@ -0,0 +1,246 @@
...
+```{r}
+sorghum_modules <- list(canopy_module_name='c4_canopy',
+                       soil_module_name='one_layer_soil_profile',
+                       growth_module_name='partitioning_growth',
+                       senescence_module_name='thermal_time_senescence',
+                       leaf_water_stress_module_name='leaf_water_stress_exponential',
+                       stomata_water_stress_module_name='stomata_water_stress_linear')
+```
+
+Use these three inputs, along with a weather file included with the package, to generate biomass values for **Setaria**.
```
Comment (Chris-Schnaufer) [inline BioCro/DARPA/biocro_1.0_darpa.Rmd:215]:
```suggestion
Use these three inputs, along with the weather file included with the package, to generate biomass values for **Setaria**. 
```

**Comment by KristinaRiemer** [comment]
I included all of your suggestions in the last commit, thanks @Chris-Schnaufer!

**Comment by KristinaRiemer** [comment]
@dlebauer normalizing the coefficients to one (see most recent commit) made the fit way worse, I'm not really sure why? 

Before normalizing (but after setting rhizome and grain coefficients to 0): 
before_normalizing

After normalizing: 
after_normalizing

File: BioCro/DARPA/biocro_1.0_darpa.Rmd [hunk 4]
```diff
@@ -0,0 +1,326 @@
...
+if(!packageVersion(pkg = 'BioCro') >= 1.0){
+  warning("need to use BioCro v1.0 or greater")
+} else {
+  #devtools::install_github('ebimodeling/biocro-dev')
+  devtools::install_github('ebimodeling/biocro@new-framework')  
+}
+
+```
+
+### Read in data
```
Comment (dlebauer) [inline BioCro/DARPA/biocro_1.0_darpa.Rmd:23]: Needs to be more clear where the data are from - this assumes I have run the biocro_biomass_darpa.Rmd script, and for that script that I have manually downloaded a csv file and put it in the root directory of the repository. 

we should be able to make use of a private place to put the data (either using googlesheets to access the current spreadsheet or using a private github repository - I prefer the private repository idea)

Comment (KristinaRiemer) [reply BioCro/DARPA/biocro_1.0_darpa.Rmd:23]: This is definitely not the most clear or reproducible due to data restrictions, so I appreciate your suggestions. I intended to use googlesheets, but UA doesn't allow me to authenticate my Google Drive. I agree with using a private repo to store the data, but are we (me?) going to have to make sure the repo is up to date with the Google Drive data?

File: BioCro/DARPA/biocro_1.0_darpa.Rmd [hunk 5]
```diff
@@ -0,0 +1,326 @@
...
+           get_growing_season_climate(OpBioGro_weather),
+           setaria_modules)
+  tt <- t %>%
+    select(TTc, Stem, Leaf, Root, Rhizome, Grain) %>%
+    rename(ThermalT = TTc)
+  ttt <- tt %>%
+    filter(round(tt$ThermalT) %in% round(OpBioGro_biomass$ThermalT))
+  bio_ests <- select(ttt, -ThermalT)
+  bio_meas <- select(OpBioGro_biomass, -ThermalT, -LAI)
+  diff <- abs(bio_ests - bio_meas)
```
[the remark this answers is not in the document]
Comment (KristinaRiemer) [reply BioCro/DARPA/biocro_1.0_darpa.Rmd:258]: Ah, yes! I did this for the old version of the model but forgot to try it with the new one, so I will do that.

File: BioCro/DARPA/biocro_1.0_darpa.Rmd [hunk 6]
```diff
@@ -0,0 +1,326 @@
...
+                       growth_module_name='partitioning_growth',
+                       senescence_module_name='thermal_time_senescence',
+                       leaf_water_stress_module_name='leaf_water_stress_exponential',
+                       stomata_water_stress_module_name='stomata_water_stress_linear')
+```
+
+### Run model 
+
+Use these three inputs, along with the weather data, to generate biomass values for **Setaria** using BioCro function `Gro`. 
+
```
[the remark this answers is not in the document]
Comment (KristinaRiemer) [reply BioCro/DARPA/biocro_1.0_darpa.Rmd:214]: I should take this out. There used to be an example there.

File: BioCro/DARPA/biocro_1.0_darpa.Rmd (hunk 4 above)
Comment (dlebauer) [reply BioCro/DARPA/biocro_1.0_darpa.Rmd:23]: yes - put in a private repo and make sure data are up to date w/ google drive. You have to download a new csv every time you make changes anyhow, so this shouldn't be that much more difficult (and would have the added benefit of version control and having everything in one place!).

Comment (KristinaRiemer) [reply BioCro/DARPA/biocro_1.0_darpa.Rmd:23]: I created [an new issue](https://github.com/danforthcenter/sentinel-detection/issues/141) for this task because it's going to apply across other vignettes in addition to this one.

File: BioCro/DARPA/biocro_1.0_darpa.Rmd [hunk 7]
```diff
@@ -0,0 +1,326 @@
...
+OpBioGro_biomass <- read.csv("biocro_opt_darpa_files/OpBioGro_biomass.csv")
+```
+
+### Set up parameters
+
+The following code uses BioCro 1.0. 
+
+Set up the parameters for *Setaria* using these two lists, `setaria_initial_state` and `setaria_parameters`. This uses parameter values from optimizing the older version of BioCro. 
+```{r, message=FALSE}
+setaria_initial_state <- with(list(), {
```
[the remark this answers is not in the document]
Comment (KristinaRiemer) [reply BioCro/DARPA/biocro_1.0_darpa.Rmd:39]: I looked a while ago to see if there is a way to shrink down big code blocks so these at least don't take up a ton of space, but didn't find anything. If these become their own files, we should discuss what the best file organizational structure would be. 

We would want to do this for all the vignettes then, like [you mentioned](https://github.com/az-digitalag/model-vignettes/pull/25#discussion_r443062453) for the temp experiment runs.

File: BioCro/DARPA/biocro_1.0_darpa.Rmd [hunk 8]
```diff
@@ -0,0 +1,326 @@
...
+  k_params_vec <- unlist(k_params)
+  k_params[10:12] <- as.list(k_params_vec[10:12]/sum(k_params_vec[10:12]))
+  k_params_vec <- unlist(k_params)
+  k_params[13:15] <- as.list(k_params_vec[13:15]/sum(k_params_vec[13:15]))
+  k_params_vec <- unlist(k_params)
+  k_params[16:19] <- as.list(k_params_vec[16:19]/sum(k_params_vec[16:19]))
+  all_params <- c(nonk_params, k_params)
+  names(all_params)[88:106] <- k_params_names
+  t <- Gro(setaria_initial_state,
+           all_params,
```
[the remark this answers is not in the document]
Comment (KristinaRiemer) [reply BioCro/DARPA/biocro_1.0_darpa.Rmd:255]: I ran `Gro` with the parameters in the order specified, and then in the rearranged order used in `opfn`. The results were the same, so it seems like the order doesn't matter.

File: BioCro/DARPA/biocro_1.0_darpa.Rmd [hunk 9]
```diff
@@ -0,0 +1,326 @@
...
+  k_params_vec <- unlist(k_params)
+  k_params[7:9] <- as.list(k_params_vec[7:9]/sum(k_params_vec[7:9]))
+  k_params_vec <- unlist(k_params)
+  k_params[10:12] <- as.list(k_params_vec[10:12]/sum(k_params_vec[10:12]))
+  k_params_vec <- unlist(k_params)
+  k_params[13:15] <- as.list(k_params_vec[13:15]/sum(k_params_vec[13:15]))
+  k_params_vec <- unlist(k_params)
+  k_params[16:19] <- as.list(k_params_vec[16:19]/sum(k_params_vec[16:19]))
+  all_params <- c(nonk_params, k_params)
+  names(all_params)[88:106] <- k_params_names
```
[the remark this answers is not in the document]
Comment (KristinaRiemer) [reply BioCro/DARPA/biocro_1.0_darpa.Rmd:253]: Specifying names doesn't seem necessary here, this must be vestigial from earlier code. But when I remove it, it produces an error downstream in the parallelization/DEoptim step that I'm not able to debug right now: 
```
Error in checkForRemoteErrors(val) : 
  8 nodes produced errors; first error: Caught exception in R_Gro_solver: Some modules required inputs variables that were not defined by the initial state, invariant parameters, varying parameters, or previous steady state modules.
Rerun the system in verbose mode for more information.
```
I'm going to punt this to [issue #144](https://github.com/danforthcenter/sentinel-detection/issues/144) for now.

File: BioCro/DARPA/biocro_1.0_darpa.Rmd [hunk 10]
```diff
@@ -0,0 +1,372 @@
...
+  soil_module_name                 = 'one_layer_soil_profile',
+  growth_module_name               = 'partitioning_growth',
+  senescence_module_name           = 'thermal_time_senescence',
+  leaf_water_stress_module_name    = 'leaf_water_stress_exponential',
+  stomata_water_stress_module_name = 'stomata_water_stress_linear')
+```
+
+### Run model 
+
+Use these three inputs, `setaria_initial_state`, `setaria_parameters` and `setaria_modules` along with the weather data, to generate biomass values for **Setaria** using BioCro function `Gro`.
```
Comment (KristinaRiemer) [inline BioCro/DARPA/biocro_1.0_darpa.Rmd:219]:
```suggestion
```

File: BioCro/DARPA/biocro_1.0_darpa.Rmd [hunk 11]
```diff
@@ -0,0 +1,371 @@
...
+ setaria_modules <- list(
+  canopy_module_name               = 'c4_canopy',
+  soil_module_name                 = 'one_layer_soil_profile',
+  growth_module_name               = 'partitioning_growth',
+  senescence_module_name           = 'thermal_time_senescence',
+  leaf_water_stress_module_name    = 'leaf_water_stress_exponential',
+  stomata_water_stress_module_name = 'stomata_water_stress_linear')
+```
+
+### Run model
```
Comment (KristinaRiemer) [inline BioCro/DARPA/biocro_1.0_darpa.Rmd:217]:
```suggestion
```
